### PR TITLE
refactor!: remove global variable 'ds' in service package

### PR DESCRIPTION
**Pull request by hahattan**
BREAKING CHANGE:
- update ProtocolDriver Initialize method signature to pass
DeviceServiceSDK interface as parameter
- update DeviceServiceSDK interface:
    - update Getter method name to be more idiomatic
    - remove Stop() method as it should only be called by SDK

this commit tried to remove the DeviceService global variable, the
RunningService() and Service() function that accessed it.

Instead, the DeviceServiceSDK interface is passed to ProtocolDriver as
a paramter in Initialize method so that developer can still access, mock and
test with it.

closes https://github.com/edgexfoundry/device-sdk-go/issues/1304

Signed-off-by: Chris Hung <chris@iotechsys.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

1. Run `device-simple` and verify it works correctly

### CODE REVIEW

**Review by ajcasagrande — CHANGES_REQUESTED** [review]
Since you are already making breaking changes, I found a few areas in the code which I feel can be cleaned up a bit.

File: pkg/models/protocoldriver.go
Comment (hahattan) [reply pkg/models/protocoldriver.go:28]: > I feel that it would make it easier to expand functionality in the future without causing breaking changes.

This is the clever sloution, updated !

File: pkg/interfaces/service.go
Comment (hahattan) [reply pkg/interfaces/service.go:25]: This was originally defined in `pkg/service/service.go` and I think we should only keep SDK implementation in that file.  
App-sdk also put this interface along with AppService interface: https://github.com/edgexfoundry/app-functions-sdk-go/blob/59d0a14a7a89f5760de83297ac9586da23f1db8b/pkg/interfaces/service.go#L72-L77, that's why I moved it here.

File: pkg/service/service.go
Comment (hahattan) [reply pkg/service/service.go:111]: It seems like IDE (gofmt) want it to be this way.  
I can confirm the help text is align in the help text.

File: example/driver/simpledriver.go
Comment (hahattan) [reply example/driver/simpledriver.go:89]: updated

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Nice Job! A few tweaks needed.

File: pkg/service/init.go
Comment (hahattan) [reply pkg/service/init.go:47]: I actually planned to move AutoEventManager to internal directory if possible because user don't need to implement it or use it. May we do this in that change ?

Comment (hahattan) [reply pkg/service/init.go]: you're right, updated.

**Review by lenny-intel — APPROVED** [review]
LGTM, once merged please update the for Device Service used in TAF to consume this.

- Device Virtual
- Device REST
- Device Modbus
- Device ONVIF Camera (@ajcasagrande FYI)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**
